src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.MethodParametersInfo.cs

```diff
@@ -0,0 +1,433 @@
+            foreach (var argument in arguments)
+            {
+                var parameterSymbol = argument.Parameter;
+                var allReferences = await SymbolFinder
+                    .FindReferencesAsync(parameterSymbol, document.Project.Solution, cancellationToken).ConfigureAwait(false);
+                // Need to check if the node is in CalleeMethodNode, because for this case
+                // void Caller() { Callee(i: 10); }
+                // void Callee(int i) { DoSomething(); }
+                // the 'i' in the caller will be considered as the referenced location
+                var allReferencedLocations = allReferences
+                    .SelectMany(@ref => @ref.Locations)
+                    .Where(location => calleeMethodNode.Contains(location.Location.FindNode(getInnermostNodeForTie: true, cancellationToken)))
+                    .ToImmutableArray();
+
+                if (allReferencedLocations.Length == 1
+                    && allReferencedLocations[0].SymbolUsageInfo.IsReadFrom())
+                {
+                    builder.Add(argument);
+                }
+            }
+
+            return builder.ToImmutableAndFree();
+        }
+
+        private TExpressionSyntax GenerateArgumentExpression(
+            SyntaxGenerator syntaxGenerator,
+            IArgumentOperation argumentOperation)
+        {
+            var parameterSymbol = argumentOperation.Parameter;
+            var argumentExpressionOperation = argumentOperation.Children.Single();
+            if (argumentOperation.ArgumentKind == ArgumentKind.ParamArray
+                && parameterSymbol.Type is IArrayTypeSymbol paramArrayParameter
+                && argumentExpressionOperation is IArrayCreationOperation arrayCreationOperation
+                && argumentOperation.IsImplicit)
+            {
+                // if this argument is a param array & the array creation operation is implicitly generated,
+                // it means it is in this format:
+                // void caller() { Callee(1, 2, 3); }
+                // void Callee(params int[] x) { }
+                // Collect each of these arguments and generate a new array for it.
+                // Note: it could be empty.
+                return (TExpressionSyntax)syntaxGenerator.AddParentheses(
+                    syntaxGenerator.ArrayCreationExpression(
+                        GenerateTypeSyntax(paramArrayParameter.ElementType, allowVar: false),
+                        arrayCreationOperation.Initializer.ElementValues.SelectAsArray(op => op.Syntax)));
+            }
+
+            // In all the other cases, one parameter should only maps to one argument.
+            if (argumentOperation.ArgumentKind == ArgumentKind.DefaultValue
+                && argumentOperation.Parameter.HasExplicitDefaultValue)
+            {
+                return (TExpressionSyntax)syntaxGenerator.LiteralExpression(argumentOperation.Parameter.ExplicitDefaultValue);
```

Comment:
ideally you use `ExpressionGenerator.GenerateExpression`.  that also works for enum member references (good to add a test for this too).